repl/src/main/scala/com/cloudera/livy/repl/SparkInterpreter.scala

```diff
@@ -298,7 +300,26 @@ class SparkInterpreter(conf: SparkConf) extends Interpreter with Logging {
                 TEXT_PLAIN -> readStdout()
               )
             case Results.Incomplete => Interpreter.ExecuteIncomplete()
-            case Results.Error => Interpreter.ExecuteError("Error", readStdout())
+            case Results.Error =>
+              def parseStdout(stdout: String): (String, Seq[String]) = {
+                stdout match {
+                  case EXCEPTION_STACK_TRACE_REGEX(ename, tracebackLines) =>
+                    var traceback = KEEP_NEWLINE_REGEX.pattern.split(tracebackLines)
+                    val interpreterFrameIdx = traceback.indexWhere(_.contains("$iwC$$iwC.<init>"))
+                    if (interpreterFrameIdx >= 0) {
+                      traceback = traceback
+                        // Remove Interpreter frames
+                        .take(interpreterFrameIdx)
+                        // Replace weird internal class name
+                        .map(_.replace("$iwC$$iwC", "UserCode"))
```

Comment:
(You could also make it a constant and avoid copy & pasting it in the test code.)